### PR TITLE
Adds an action required state to the DIFM Lite in progress screen

### PR DESCRIPTION
**Pull request by aneeshd16**
#### Changes proposed in this Pull Request

This will be merged into the parent PR: https://github.com/Automattic/wp-calypso/pull/59895

After checkout, we take users directly to the “Website Content” form. If they do not fill it at that time, we need to give them a path to return to the form. Once the user has paid for the product, we block access to Calypso pages by showing an “In-Progress” screen.

This page will have two states:
* Website Content not submitted: This should have a link to the Website Content form. (User action required) (Implemented in this PR)
* Website Content submitted: No user action is required here. (This was done via https://github.com/Automattic/wp-calypso/pull/60881)

#### Testing instructions

* Purchase DIFM Lite by proceeding through the flow at /start/do-it-for-me.
* After checkout, do not fill the Website Content form. Instead go to `/home/<site slug>`, where `<site slug>` is the site for which the purchase was made.
* Confirm that the screen looks like this:
image
* Click on the "Add content to your website" button and fill out the form.
* If the form has been successfully submitted, you should see this screen:
image

### CODE REVIEW

File: client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx [hunk 1]
```diff
@@ -54,24 +59,43 @@ function DIFMLiteInProgress( { siteId }: DIFMLiteInProgressProps ): React.ReactE
...
+			action={ translate( 'Add content to your website' ) }
+			actionURL={ getWebsiteContentLink( slug ) }
+			secondaryActionCallback={ recordEmailClick }
+			illustration={ WebsiteContentRequiredIllustration }
+			illustrationWidth={ 144 }
+			className="difm-lite-in-progress__content"
+		/>
+	);
+
+	return isWebsiteContentSubmitted ? renderUserActionNotRequired() : renderUserActionRequired();
```
Comment (ddc22) [inline client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx]: Why not just return the element in the ternary without calling functions? The flag variable is perfectly named for this to be understandable :)

File: client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx [hunk 2]
```diff
@@ -54,24 +59,43 @@ function DIFMLiteInProgress( { siteId }: DIFMLiteInProgressProps ): React.ReactE
...
-				secondaryAction={ hasEmailWithUs ? translate( 'Manage email' ) : translate( 'Add email' ) }
-				secondaryActionURL={ emailManagement( slug, null ) }
-				secondaryActionCallback={ recordEmailClick }
-				illustration={ SiteBuildInProgressIllustration }
-				illustrationWidth={ 144 }
-				className="difm-lite-in-progress__content"
-			/>
-		</div>
+	// No need to check for null here since the button is not rendered till we have a slug
+	const getWebsiteContentLink = ( slug: SiteSlug | null ) =>
```
Comment (ddc22) [inline client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx]: Is this function required? Can't we just include this in the component itself as a prop?

File: client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx (hunk 1 above)
Comment (ddc22) [reply client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx]: If the "options" get late to load over the wire, this screen shows the content not submitted view.
Shall we check for strong equality for "true" and "false" and if not return a null or a loader if not?

**Review by ddc22 — APPROVED** [review]
Have some minor comments, but nothing serious!
Can be merged after resolving them.
The change works as described!  🚢

File: client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx (hunk 1 above)
Comment (aneeshd16) [reply client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx]: We already show a placeholder if the site data has not been loaded: https://github.com/Automattic/wp-calypso/blob/ca0c247db82a2ec90082a64e7be5896ba8b6dd71/client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx#L39-L43

`isDIFMLiteWebsiteContentSubmitted` checks for keys within the site object itself, so the placeholder should work correctly. However, the illustration shown during the loading phase is `SiteBuildInProgressIllustration`, which is not exactly ideal. What do you think?
Kapture 2022-02-14 at 18 48 26

**Comment by aneeshd16** [comment]
> Have some minor comments, but nothing serious! Can be merged after resolving them. The change works as described! 🚢

Thanks for your feedback, I have added the fixes. Regarding the placeholder issue, I added a comment here: https://github.com/Automattic/wp-calypso/pull/61070#discussion_r805832557.

I am merging this for now, and we can tackle any issues in a separate PR. 👍

File: client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx (hunk 1 above)
Comment (ddc22) [reply client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx]: Oh ok, strange that I saw that `content not submitted` view on one instance of loading this screen which switched to the `we are building your site` view moments later 🤔

Comment (aneeshd16) [reply client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx]: You're right, @mikeicode noticed the same thing here: pdh1Xd-xv-p2#comment-832

I'll investigate this and fix this ASAP.

**Pull request merged**